### PR TITLE
Ask for a build to fetch tests (jenkins)

### PR DESCRIPTION
**Pull request by JoseSilvaHermida**
To fetch the tests for Jenkins, either the --builds or the --last-build arguments are required.

Refactored the method Jenkins.get_tests() to fetch the builds using the Jenkins.get_builds() method instead of custom queries.
Added new MissingArgument exception

### CODE REVIEW

**Review by adrianfusco — APPROVED** [review]
I've tested it with all `--builds`, with specific `--builds 13`, and `--last-build` of a job and it works really fine. Also tested when we don't give any build options. 

+1

File: cibyl/sources/jenkins.py [hunk 1]
```diff
@@ -318,62 +316,34 @@ def get_tests(self, **kwargs):
...
             :rtype: :class:`AttributeDictValue`
         """
 
-        # Get the jobs with the last completed build (default behavior)
-        jobs_found = self.send_request(
-            self.jobs_last_completed_build_query)["jobs"]
-        jobs_filtered = filter_jobs(jobs_found, **kwargs)
-
-        job_objects = {}
+        if not kwargs.get('builds') and not kwargs.get('last_build'):
```
Comment (cescgina) [inline cibyl/sources/jenkins.py:324]: Let's make sure to reuse this code, for example moving it to a method of the Source class in all sources that implement get_tests.

**Review by cescgina — APPROVED** [review]
Looks good

File: cibyl/sources/jenkins.py (hunk 1 above)
Comment (adrianfusco) [reply cibyl/sources/jenkins.py:324]: > Let's make sure to reuse this code, for example moving it to a method of the Source class in all sources that implement get_tests.

As we talk recently, we will merge these changes, check the Elasticsearch changes and merge it when it's done and then we will move the method to implement in all `get_tests`.

Thanks!

**Pull request merged**